File: app/src/main/java/ru/vtb/ppcd/example/processor/impl/BasicErrorHandler.java
package ru.vtb.ppcd.example.processor.impl;

import ru.vtb.ppcd.example.processor.ErrorHandler;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.AllArgsConstructor;
import lombok.SneakyThrows;
import lombok.extern.slf4j.Slf4j;
import org.apache.camel.Exchange;
import org.springframework.stereotype.Component;
import ru.vtb.ppcd.generated.async.model.Error;

@Component
@AllArgsConstructor
@Slf4j
public class BasicErrorHandler implements ErrorHandler {
    private final ObjectMapper objectMapper;

    @SneakyThrows
    @Override
    public void processMessage(Exchange exchange, Exception exception) {
        var error = new Error()
                .message(exception.getLocalizedMessage())
                .code(Error.CodeEnum.INTERNAL_SERVER_ERROR);

        var outcomeMessage = objectMapper.writeValueAsString(error);

        exchange.getIn().setBody(outcomeMessage);

        log.error("Error occurred {}", exchange.getMessage(), exception);
    }
}

File: ms-ppcd-example-app/src/main/kotlin/ru/vtb/ppcd/example/processor/impl/BasicErrorHandler.kt
package org.example.ru.vtb.ppcd.example.processor.impl

import com.fasterxml.jackson.databind.ObjectMapper
import io.github.oshai.kotlinlogging.KotlinLogging
import org.apache.camel.Exchange
import org.example.ru.vtb.ppcd.example.processor.ErrorHandler
import org.springframework.stereotype.Component
import ru.vtb.ppcd.generated.async.model.Error

private val logger = KotlinLogging.logger {}

@Component
class BasicErrorHandler(
    private val objectMapper: ObjectMapper
) : ErrorHandler {
    override fun processMessage(exchange: Exchange, exception: Exception) {
        Error(
            message = exception.message,
            code = Error.Code.INTERNAL_SERVER_ERROR
        ).apply {
            val outcomeMessage = objectMapper.writeValueAsString(this);
            exchange.`in`.body = outcomeMessage

            logger.error {
                "BasicErrorHandler.processMessage: произошла ошибка - " +
                        "${exchange.getMessage()}, ex - $exception"
            }
        }
    }
}